include/onnxruntime/core/framework/execution_provider.h

```diff
@@ -60,11 +60,15 @@ enum class DataLayout {
 class IExecutionProvider {
  protected:
   IExecutionProvider(const std::string& type, bool use_metadef_id_creator = false)
-      : type_{type} {
+      : IExecutionProvider(type, OrtDevice(), use_metadef_id_creator) {}
+
+  IExecutionProvider(const std::string& type, OrtDevice device, bool use_metadef_id_creator = false)
+      : default_device_(device), type_{type} {
     if (use_metadef_id_creator) {
       metadef_id_generator_ = std::make_unique<ModelMetadefIdGenerator>();
     }
   }
+  OrtDevice default_device_;
```

Comment:
this should be private?

and please add comments for the new member/method.

Reply:
it is protected. 

Comments added